zcash_client_backend/src/data_api.rs

```diff
@@ -44,6 +47,95 @@ pub enum NullifierQuery {
     All,
 }
 
+#[derive(Debug, Clone)]
+pub struct AccountBalance {
+    pub spendable_value: NonNegativeAmount,
+    pub pending_change_value: NonNegativeAmount,
+    pub value_pending_spendability: NonNegativeAmount,
+    pub unshielded: NonNegativeAmount,
```

Comment:
The primary reason that I made the struct fields public is that it's convenient to be able to mutate them in https://github.com/zcash/librustzcash/pull/914/commits/1e5eae87a4412bb66789f563e64aa9905e512261#diff-a1724f7547ecabc613a07520eaf7917d9084d4277219d186698b0c8e352495a4R643-R649, and since the balances themselves are only available immutably via reference from `WalletSummary`, I figured that was okay.

Reply:
I'll add extra documentation to the `unshielded` field to the following effect:

Unshielded balances are not subject to confirmation-depth constraints, because the only possible operation on a transparent balance is to shield it, it is possible to create a zero-conf transaction to perform that shielding, and the resulting shielded notes will be subject to normal confirmation rules. So there's currently no reason to make transparent funds subject to confirmation requirements at all.

Some of the other existing APIs should probably be updated to reflect this realization.

Reply:
Right, we don't have any automatic shielding functionality at all, so at present this isn't a concern.

Reply:
Right, but defining this to include zero-conf funds means that an adversary will be able to arbitrarily inflate it above the realistic balance, as soon as we implement detection of zero-conf funds.

If I send you funds in a bunch of transactions that each have 50 unpaid actions and expire at the next block, I can be pretty confident that at most one (probably zero) of them will be mined.